js/modules/loki_message_api.js

```diff
@@ -77,6 +78,48 @@ class LokiMessageAPI {
   async sendMessage(pubKey, data, messageTimeStamp, ttl, options = {}) {
     const { isPing = false, numConnections = DEFAULT_CONNECTIONS } = options;
     // Data required to identify a message in a conversation
+
+    if (pubKey.match(/^06/)) {
+      // console.log('HIJACK', data);
+      // console.log('HIJACK dataMessage', data.message.dataMessage);
+      const payload = {
+        text: data.message.dataMessage.body,
+        annotations: [
+          {
+            type: 'network.loki.messenger.publicChat',
+            value: {
+              // id: data.id,
+              // channel will know this
+              // group: data.group,
+              timestamp: messageTimeStamp,
+              from: data.ourKey,
+            },
+          },
+        ],
+      };
+      // FIXME: use data.group to figure out channel id
+      // const res =
+      await nodeFetch('https://chat.lokinet.org/channels/1/messages', {
+        method: 'post',
+        headers: {
+          // x-www-form-urlencoded
+          'Content-Type': 'application/json',
+          Authorization: 'Bearer loki',
+        },
+        body: JSON.stringify(payload),
+      });
+      // const body = await res.text();
+
+      // Causes "Error setting p2p on message"
```

Comment:
Not sure if we want to mark these messages as p2p?
If we do tho `data.group` would be referencing the protobuf `GroupContext` right?
Also from below the name of the field it is expecting is `pubKey`, not `pubkey`
And should this not be pubKey?

At any rate these comments should either be removed or the messages actually marked as p2p

Reply:
no data.group was the group's pubKey (not sure what a `GroupContext` is)

>Also from below the name of the field it is expecting is pubKey, not pubkey And should this not be pubKey?

Correct

>At any rate these comments should either be removed or the messages actually marked as p2p
It's a WIP, only posted to see if you can figure out why messages aren't being saved.